services/app-api/src/handlers/reports.ts

```diff
@@ -0,0 +1,116 @@
+    formDataProto: HealthPlanFormDataType | Error
+    id: string
+    createdAt: Date
+    pkgID: string
+    submittedAt: Date | null
+    unlockedAt: Date | null
+    unlockedBy: string | null
+    unlockedReason: string | null
+    submittedBy: string | null
+    submittedReason: string | null
+    programNames?: string[]
+}
+
+/* formProtoData is an encoded protocal buffer in the db,
+so we decode it and put it back on the revision as readable data */
+const decodeRevisions = (
+    revisions: HealthPlanRevisionTable[]
+): RevisionWithDecodedProtobuf[] => {
+    return revisions.map((revision) => {
+        return {
+            ...revision,
+            formDataProto: toDomain(revision.formDataProto),
+        }
+    })
+}
+
+/* We only store the program IDs in the db,
+so we look up the program names and add them to the revision */
+const decorateRevisionsWithProgramNames = (
+    revisions: RevisionWithDecodedProtobuf[]
+) => {
+    const programList = [] as ProgramArgType[]
+    statePrograms.states.forEach((state) => {
+        programList.push(...state.programs)
+    })
+    revisions.forEach((revision) => {
+        const names = programList
+            .filter(
+                (p) =>
+                    !(revision.formDataProto instanceof Error) &&
+                    revision.formDataProto.programIDs.includes(p.id)
+            )
+            .map((p) => p.name)
+        revision.programNames = names
+    })
+    return revisions
+}
+
+export const main: APIGatewayProxyHandler = async () => {
+    const dbURL = process.env.DATABASE_URL
+    const secretsManagerSecret = process.env.SECRETS_MANAGER_SECRET
+    if (!dbURL) {
+        throw new Error('Init Error: DATABASE_URL is required to run app-api')
+    }
+
+    const pgResult = await configurePostgres(dbURL, secretsManagerSecret)
+    if (pgResult instanceof Error) {
+        console.error(
+            "Init Error: Postgres couldn't be configured in data exporter"
+        )
+        throw pgResult
+    }
+
+    const store = NewPostgresStore(pgResult)
+    const result: HealthPlanRevisionTable[] | StoreError =
+        await store.getAllRevisions()
+    if (isStoreError(result)) {
+        throw new Error('Error getting records; cannot generate report')
+    }
+    const allDecodedRevisions: RevisionWithDecodedProtobuf[] =
+        decorateRevisionsWithProgramNames(decodeRevisions(result))
+
+    const bucket = [] as RevisionWithDecodedProtobuf[]
+    allDecodedRevisions.forEach((revision) => {
+        if (revision.formDataProto instanceof Error) {
+            throw new Error(`Error generating reports array`)
+        } else {
+            bucket.push(revision)
+        }
+    })
+
+    const parser = new Parser({
+        transforms: [
+            transforms.flatten({
+                objects: true,
+                arrays: false,
+                separator: ',',
+            }),
+        ],
+    })
+    const csv = await parser.parse(bucket)
+
+    return {
```

Comment:
suggestion: consider returning error code for the errors or capture the errors and console.log them in the handler (those logs go to cloudwatch). 

I remember doing that for the email submit lambda because otherwise I was noticing the fail reason was difficult to find if you just throw. I could be off though ... maybe you had no issue with this.

Reply:
oops just ran through this on deploy app and getting uncaught error when clicking link. can't see specific message but its in my dev tools console

Reply:
I don't have the review app cms user in my password vault, so I'll take a look when we can get that sorted out...